src/match/controllers/update_match_controller.py

```diff
@@ -0,0 +1,131 @@
+            if existing_location.id not in body_proposed_locations:
+                self._match.proposed_locations.remove(existing_location)
+
+    def _update_meeting_location(self, body_meeting_location_id):
+        """Update the meeting location."""
+        new_meeting_location = Location.objects.filter(id=body_meeting_location_id)
+        if not new_meeting_location:
+            return failure_response(
+                f"Location with id {body_meeting_location_id} does not exist."
+            )
+        self._modify_attribute("meeting_location", new_meeting_location[0])
+
+    def _update_meeting_time(self, times):
+        """Given a list with one time, generate a timestamp.
+        Example: ["", "", "9.5", "", "", "", ""] turns into
+        '2021-05-11 09:30:00' i.e. the Tuesday after now() at 9:30 am
+        """
+        # simple_meeting is [weekday, time] from times param
+        simple_meeting = [0, 0]
+        for index in range(len(times)):
+            if times[index] != "":
+                simple_meeting = index, float(times[index])
+        now = datetime.datetime.now()
+        hour, minute = divmod(simple_meeting[1], 1)
+        minute *= 60
+        days = (6 - now.weekday() + simple_meeting[0]) % 7
+        meeting_time = (now + datetime.timedelta(days=days)).replace(
+            hour=int(hour), minute=int(minute), second=0, microsecond=0
+        )
+        self._match.meeting_time = meeting_time
+
+    def _update_match_status(
+        self, proposed_meeting_times, proposed_locations, meeting_location, meeting_time
+    ):
+        """Update status based on match field changes."""
+        if proposed_meeting_times is not None and proposed_locations is not None:
+            # In case match is already proposed
+            if self._match.status == match_status.PROPOSED:
+                return failure_response(
+                    "Match has already been proposed", status.HTTP_400_BAD_REQUEST
+                )
+            self._match.status = match_status.PROPOSED
+            self._match.proposer_id = self._user.id
+            self._match.accepted_ids = [self._user.id]
+        elif meeting_location is not None and meeting_time is not None:
+            # In case match is already active
+            if self._match.status == match_status.ACTIVE:
+                return failure_response(
+                    "Match has already been accepted", status.HTTP_400_BAD_REQUEST
+                )
+            self._match.status = match_status.ACTIVE
+            # use json.loads to convert a string of a list to an actual list
+            accepted_ids = json.loads(self._match.accepted_ids)
+            if self._user.id not in accepted_ids:
+                accepted_ids.append(self._user.id)
+                self._match.accepted_ids = accepted_ids
+            self._match.proposed_meeting_times = None
+            self._match.proposed_locations.set([])
```

Comment:
why do you need to do this? can't you just leave `self._match.proposed_locations` alone?

Reply:
in our original call with frontend you mentioned that when state changes we should update non-necessary fields to None or empty set, lmk if I don't need to do that tho